subsys/bluetooth/host/monitor.c

```diff
@@ -74,18 +64,100 @@ static struct {
 	atomic_t other;
 } drops;
 
+static void drop_add(uint16_t opcode)
+{
+	switch (opcode) {
+	case BT_MONITOR_COMMAND_PKT:
+		atomic_inc(&drops.cmd);
+		break;
+	case BT_MONITOR_EVENT_PKT:
+		atomic_inc(&drops.evt);
+		break;
+	case BT_MONITOR_ACL_TX_PKT:
+		atomic_inc(&drops.acl_tx);
+		break;
+	case BT_MONITOR_ACL_RX_PKT:
+		atomic_inc(&drops.acl_rx);
+		break;
+#if defined(CONFIG_BT_BREDR)
+	case BT_MONITOR_SCO_TX_PKT:
+		atomic_inc(&drops.sco_tx);
+		break;
+	case BT_MONITOR_SCO_RX_PKT:
+		atomic_inc(&drops.sco_rx);
+		break;
+#endif
+	default:
+		atomic_inc(&drops.other);
+		break;
+	}
+}
+
+#if defined(CONFIG_BT_DEBUG_MONITOR_RTT)
+#include <SEGGER_RTT.h>
+
+#define RTT_BUFFER_NAME CONFIG_BT_DEBUG_MONITOR_RTT_BUFFER_NAME
+#define RTT_BUF_SIZE CONFIG_BT_DEBUG_MONITOR_RTT_BUFFER_SIZE
+
+/* Intermediate buffer */
+static uint8_t rtt_buf[RTT_BUF_SIZE];
+static size_t rtt_buf_offset;
+
+static void monitor_send(const void *data, size_t len)
+{
+	struct bt_monitor_hdr *hdr = (struct bt_monitor_hdr *)rtt_buf;
+	unsigned int cnt = 0;
+	bool drop;
+
+	/* Drop any packet which cannot fit the buffer */
+	drop = rtt_buf_offset + len > sizeof(rtt_buf);
+	if (!drop) {
+		memcpy(rtt_buf + rtt_buf_offset, data, len);
```

Comment:
```suggestion
		(void)memcpy(rtt_buf + rtt_buf_offset, data, len);
```

Reply:
OK, but it's overkill. I've never saw anyone making use of memcpy return

Reply:
It is just convention in Zephyr (and technically part of MISRA, but we are not enforcing that (yet)). 

We do actually also have this in Zephyr: 
```c
	bufs->buffers = memcpy(buf_copy,
			       bufs->buffers,
			       bufs->count * sizeof(struct spi_buf));
``` 
But otherwise I agree that it's very rare that the return pointer is used.